fe/src/hooks/useCategorySelector.ts

```diff
@@ -0,0 +1,40 @@
+import { useCategories } from '@/queries/category';
+import { useEffect, useState } from 'react';
+
+type CategorySelectorType = {
+  initialCategoryName?: string;
+};
+
+export const useCategorySelector = ({
+  initialCategoryName,
+}: CategorySelectorType) => {
+  const [selectedCategory, setSelectedCategory] = useState<CategoryType>();
+  const { categories } = useCategories();
+
+  useEffect(() => {
+    if (!categories || !initialCategoryName) {
+      return;
+    }
+
+    const initialCategory = categories.find(
+      (category) => category.name === initialCategoryName,
+    )!;
+
+    setSelectedCategory(initialCategory);
+  }, [categories, initialCategoryName]);
```

Comment:
명시적으로 인자를 넘기고 싶어서 객체를 전달받게 했는데, 사용처의 모습이 별로네요.
```
const { selectedCategory, categories, selectCategory } = useCategorySelector({});
```

인자 기본값으로 빈 객체를 할당하면 어떻까요?

```suggestion
export const useCategorySelector = ({
  initialCategoryName,
}: CategorySelectorType = {}) => {
  const [selectedCategory, setSelectedCategory] = useState<CategoryType>();
  const { categories } = useCategories();

  useEffect(() => {
    if (!categories || !initialCategoryName) {
      return;
    }

    const initialCategory = categories.find(
      (category) => category.name === initialCategoryName,
    )!;

    setSelectedCategory(initialCategory);
  }, [categories, initialCategoryName]);
```

Reply:
저는 괜찮은 아이디어라고 생각합니다~!